src/test/functional/sproc.spec.ts

```diff
@@ -5,181 +5,172 @@ import {
-            } catch (err) {
-                throw err;
-            }
-        });
+            const { result: sproc } = await container.storedProcedures.create(sprocDefinition);
 
-        it("nativeApi Should do sproc CRUD operations successfully rid based", async function () {
+            assert.equal(sproc.id, sprocDefinition.id);
+            assert.equal(sproc.body, "function () { const x = 10; }");
+
+            // read sprocs after creation
+            const { result: sprocsAfterCreation } = await container.storedProcedures.read().toArray();
+            assert.equal(sprocsAfterCreation.length, beforeCreateSprocsCount + 1, "create should increase the number of sprocs");
+
+            // query sprocs
+            const querySpec = {
+                query: "SELECT * FROM root r",
+            };
+            const { result: queriedSprocs } = await container.storedProcedures.query(querySpec).toArray();
+            assert(queriedSprocs.length > 0, "number of sprocs for the query should be > 0");
+
+            // replace sproc
+            sproc.body = function () { const x = 20; };
+            const { result: replacedSproc } = await container.storedProcedures.getStoredProcedure(sproc.id).replace(sproc);
+
+            assert.equal(replacedSproc.id, sproc.id);
+            assert.equal(replacedSproc.body, "function () { const x = 20; }");
+
+            // read sproc
+            const { result: sprocAfterReplace } = container.storedProcedures.getStoredProcedure(replacedSproc.id).read();
+            assert.equal(replacedSproc.id, sprocAfterReplace.id);
+
+            // delete sproc
+            await container.storedProcedures.getStoredProcedure(replacedSproc.id).delete();
+
+            // read sprocs after deletion
             try {
-                await sprocCRUDTest(false, false);
+                await container.storedProcedures.getStoredProcedure(replacedSproc.id).read();
+                assert.fail("Must fail to read sproc after deletion");
             } catch (err) {
-                throw err;
+                const notFoundErrorCode = 404;
+                assert.equal(err.code, notFoundErrorCode, "response should return error code 404");
             }
         });
 
         it("nativeApi Should do sproc CRUD operations successfully name based with upsert", async function () {
-            try {
-                await sprocCRUDTest(true, true);
-            } catch (err) {
-                throw err;
-            }
-        });
+                   // read sprocs
+            const { result: sprocs } = await container.storedProcedures.read().toArray();
+            assert.equal(sprocs.constructor, Array, "Value should be an array");
+
+            // create a sproc
+            const beforeCreateSprocsCount = sprocs.length;
+            const sprocDefinition: StoredProcedureDefinition = {
+                id: "sample sproc",
+                body() { const x = 10; },
```

Comment:
This test if failing but I think it's a typescript thing, not a new SDK thing. We should think about pulling these files in as raw content instead of inlining them in typescript. We don't want tsc to touch them.